### PR TITLE
Setup Wizard: Theme Style Screen

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add a screen to the Setup Wizard for selecting which theme (Newspack or one of the 5 Child Themes) to start with. Each theme can be previewed in the relatively recent `WebPreview` component and activated from there. The preview sites are:

- https://newspack.newspackstaging.com/
- https://nelson.newspackstaging.com/
- https://katharine.newspackstaging.com/
- https://joseph.newspackstaging.com/
- https://scott.newspackstaging.com/
- https://sacha.newspackstaging.com/

The PR introduces three new components: `ButtonGroup`, `StyleCard` and `StyleCardGroup`. The Webpack changes are here because this is the first case of raster assets being included in components. 

Screen Shot 2020-01-29 at 2 39 41 PM

Screen Shot 2020-01-29 at 2 39 48 PM

### How to test the changes in this Pull Request:

1. After switching to this branch and building, navigate to the Setup Wizard (Settings->Newspack then click link at the bottom)
2. Advance to the sixth page: Theme
3. If you already have a Newspack theme activated, verify it is marked "Active."
4. Try Previewing and Activating each theme. Verify the preview looks good, and that the activation is successful. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by jeffersonrabb** [comment]
@thomasguillot I rebased this and added support for child theme installation in https://github.com/Automattic/newspack-plugin/pull/364/commits/b6bf348f664226de80df8fc1b53feef7aa8508bc. Now that we're using Child Themes I was thinking about refactoring all the class and function names to reflect this, e.g. `theme-style-selection` -> `child-theme-selection`. Any objection to this?

**Comment by thomasguillot** [comment]
No objections from me.

**Comment by jeffersonrabb** [comment]
> No objections from me.

Done in https://github.com/Automattic/newspack-plugin/pull/364/commits/ded480ea4b5e2c1559b4325197b7fe97e2432190

File: assets/components/src/web-preview/index.js [hunk 1]
```diff
@@ -48,15 +48,6 @@ class WebPreview extends Component {
 		}
 	}
 
-	/**
```
Comment (jeffersonrabb) [inline assets/components/src/web-preview/index.js]: FYI @adekbadek I had to remove the `componentWillUnmount` after finding that exiting a view with multiple `WebPreview` components caused a crash. If there are multiple instances of the component, the first one to unmount will remove the DOM element leaving the others with a reference to a non-existent element. When the other unmount this throws an error. Also,  we wouldn't want a situation where one `WebPreview` unmounts and removes the DOM element that is still in being used by another. All this means is we may be left  with a single  empty div sitting around, which isn't really so bad...

Comment (adekbadek) [reply assets/components/src/web-preview/index.js]: Yeah, I guess in this case it's ok to skip the cleanup

**Comment by jeffersonrabb** [comment]
@thomasguillot the 6 demo sites have been created and in https://github.com/Automattic/newspack-plugin/pull/364/commits/b337c826d48b03e3abb4c60ee482fd39dbe2efb8 I changed the preview URLs. I assume you'll want to make new thumbnails based on these?

**Comment by thomasguillot** [comment]
@jeffersonrabb yes. The screenshots should match the demo sites.

**Comment by thomasguillot** [comment]
> @thomasguillot the 6 demo sites have been created and in [b337c82](https://github.com/Automattic/newspack-plugin/commit/b337c826d48b03e3abb4c60ee482fd39dbe2efb8) I changed the preview URLs. I assume you'll want to make new thumbnails based on these?

Done in https://github.com/Automattic/newspack-plugin/pull/364/commits/468253142f2bdfd8c9cdd189f5d07d71211d0b4d

File: assets/components/src/style-card/index.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	 */
+	render( props ) {
+		const { className, cardTitle, url, image, isActive, onClick, ...otherProps } = this.props;
+		const classes = classnames(
+			'newspack-style-card',
+			isActive && 'newspack-style-card__is-active',
+			className
+		);
+		return (
+			<div className={ classes } tabindex="0">
```
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply assets/components/src/style-card/index.js]: Fixed in https://github.com/Automattic/newspack-plugin/pull/364/commits/b75e2f25be65cdcfd53bc45ba634a0693a4c6fd0

File: assets/components/src/button-group/index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	 * Render.
+	 */
+	render( props ) {
+		const { className, ...otherProps } = this.props;
+		const classes = classnames( 'newspack-button-group', className );
+		return <BaseComponent className={ classes } { ...otherProps } />;
+	}
+}
+
+export default ButtonGroup;
```
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply assets/components/src/button-group/index.js]: Looks like you're right - I think this was from an earlier iteration before we switched to `StyleCard` and `StyleCardGroup`. Removed in https://github.com/Automattic/newspack-plugin/pull/364/commits/97c8b5d4b229dbef98a702860b54942938aba864

File: assets/components/src/style-card/index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Style Card
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { Component } from '@wordpress/element';
+// import { DropdownMenu } from '@wordpress/components';
```
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply assets/components/src/style-card/index.js]: Removed in https://github.com/Automattic/newspack-plugin/pull/364/commits/f5e38a378b7a111f706b3d9141e681d8b6021157

File: assets/wizards/setup/views/theme-selection/index.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import NewspackImg from './images/newspack.png';
+import ScottImg from './images/scott.png';
+import NelsonImg from './images/nelson.png';
+import KatharineImg from './images/katharine.png';
+import SachaImg from './images/sacha.png';
+import JosephImg from './images/joseph.png';
+
+/**
+ * Theme Selection Screen.
+ */
```
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply assets/wizards/setup/views/theme-selection/index.js]: https://github.com/Automattic/newspack-plugin/pull/364/commits/52b427fe3db672a371071740adb7988235a1e757

File: assets/components/src/style-card/style.scss [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    }
+  }
+
+  // Web Preview
+  body.newspack-web-preview__open & {
+    .newspack-style-card__actions {
+      background: transparent;
+      opacity: 1;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply assets/components/src/style-card/style.scss]: https://github.com/Automattic/newspack-plugin/pull/364/commits/52b427fe3db672a371071740adb7988235a1e757

File: assets/wizards/setup/views/theme-selection/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (jeffersonrabb) [reply assets/wizards/setup/views/theme-selection/index.js]: I was under the impression there was supposed to be a JSDoc comment just above the class definition, is this incorrect? (You'll find many similar duplicates throughout the codebase 😬) Removed in https://github.com/Automattic/newspack-plugin/pull/364/commits/6a4e1b385970caf074fbb8f9ca743cc96e194cb5

**Comment by jeffersonrabb** [comment]
Thanks for reviewing @adekbadek, I think I addressed everything.

File: assets/wizards/setup/views/theme-selection/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (adekbadek) [reply assets/wizards/setup/views/theme-selection/index.js]: Yes - they should be just above the class definition. I meant that the other one (at the beginning of the file) should be removed :D

**Pull request merged**